widgetsnbextension/src/manager.js

```diff
@@ -219,6 +219,24 @@ export class WidgetManager extends ManagerBase {
    */
   _init_actions() {
     var notifier = Jupyter.notification_area.widget('widgets');
+    this.notebook.events.on('before_save.Notebook', async () => {
+      var cells = Jupyter.notebook.get_cells();
+      // notebook.js save_notebook doesn't want for this promise, we are simply lucky when this
+      // finishes before saving.
```

Comment:
This seems very weak :S

Reply:
Yes, I think this means it will be a lab feature only

Reply:
What about not making it a Promise?

Reply:
we to await the model, and making `generateMimeBundle async also makes sense, since it will involve async cases in bqplot at least.